src/webui/webapplication.cpp

```diff
@@ -700,11 +721,40 @@ bool WebApplication::validateHostHeader(const QStringList &domains) const
     }
 
     LogMsg(tr("WebUI: Invalid Host header. Request source IP: '%1'. Received Host header: '%2'")
-           .arg(m_env.clientAddress.toString(), m_request.headers[Http::HEADER_HOST])
+           .arg(m_clientAddress.toString(), m_request.headers[Http::HEADER_HOST])
             , Log::WARNING);
     return false;
 }
 
+QHostAddress WebApplication::resolveClientAddress() const
+{
+    QHostAddress clientAddress = m_env.clientAddress;
+
+    if (!m_isReverseProxySupportEnabled)
+        return clientAddress;
+
+    // Only reverse proxy can overwrite client address
+    if (m_trustedReverseProxyList.contains(clientAddress))
```

Comment:
I suggest you use:
```c++
if (!m_trustedReverseProxyList.contains(clientAddress))
    return m_env.clientAddress;
```
and move `QHostAddress clientAddress` further below, near to where it is used.